include/boost/static_string/static_string.hpp

```diff
@@ -2608,6 +2608,108 @@ class basic_static_string
     const_iterator first,
     const_iterator last);
 
+  /** Erase from the string.
+
+      Erases all the characters from the string satisfying specific criteria.
+
+      @par Exception Safety
+
+      Strong guarantee.
+
+      @note All references, pointers, or iterators
+      referring to contained elements are invalidated. Any
+      past-the-end iterators are also invalidated.
+
+      @return `*this`
+
+      @param pred A unary predicate which returns ​true if the element should be erased.
+  */
+  template<typename UnaryPredicate>
+  BOOST_STATIC_STRING_CPP14_CONSTEXPR
+  basic_static_string&
+  erase_if(UnaryPredicate pred)
+  {
+    erase_if(data(), end(), pred);
+    return *this;
+  }
+
+  /** Erase from the string.
+
+      Erases `num` characters from the string, starting at `index`, satisfying specific criteria.
+      `num` is determined as the smaller of `count` and `size() - index`.
+
+      @par Exception Safety
+
+      Strong guarantee.
+
+      @note All references, pointers, or iterators
+      referring to contained elements are invalidated. Any
+      past-the-end iterators are also invalidated.
+
+      @return `*this`
+
+      @param index The index to erase at.
+      The default argument for this parameter is `0`.
+
+      @param count The number of characters to erase.
+      The default argument for this parameter is @ref npos.
+
+      @throw std::out_of_range `index > size()`
+  */
+  template<typename UnaryPredicate>
+  BOOST_STATIC_STRING_CPP14_CONSTEXPR
+  basic_static_string&
+  erase_if(
+    size_type index,
+    size_type count,
+    UnaryPredicate pred)
+  {
+    erase_if(data() + index, data() + index + capped_length(index, count), pred);
+    return *this;
+  }
+
+  /** Erase from the string.
+
+      Erases the characters satisfying specific criteria in the range `{first, last)`.
+
+      @par Precondition
+
+      `{first, last}` shall be valid within `{data(), data() + size()}`
+
+      @par Exception Safety
+
+      Strong guarantee.
+
+      @tparam UnaryPredicate The type of the predicate.
+
+      @par Constraints
+
+      `UnaryPredicate` satisfies __UnaryPredicate__.
+
+      @note All references, pointers, or iterators
+      referring to contained elements are invalidated. Any
+      past-the-end iterators are also invalidated.
+
+      @return An iterator referring to the character `last`
+      previously referred to, or @ref end() if one does not exist.
+
+      @param first An iterator referring to the first character to erase.
+
+      @param last An iterator past the last character to erase.
+
+      @param pred A unary predicate which returns ​true if the element should be erased.
+  */
+  template<
+    typename ForwardIt,
+    typename UnaryPredicate,
+    typename std::enable_if<detail::is_forward_iterator<ForwardIt>::value>::type* = nullptr>
```

Comment:
this needs to be ifdeffed for the docs to excclude it, and the javadoc needs a "@par Constraints" section.

Reply:
Done